daphne_worker/src/config.rs

```diff
@@ -340,6 +336,28 @@ pub(crate) struct DaphneWorkerState {
 
     /// Task list.
     tasks: Arc<RwLock<HashMap<Id, DapTaskConfig>>>,
+}
+
+impl DaphneWorkerSharedState {
+    pub(crate) fn from_worker_env(env: &Env) -> Result<Self> {
+        let config = DaphneWorkerConfig::from_worker_env(env)?;
+
+        Ok(Self {
+            config,
+            hpke_receiver_configs: Arc::new(RwLock::new(HashMap::new())),
+            leader_bearer_tokens: Arc::new(RwLock::new(HashMap::new())),
+            collector_bearer_tokens: Arc::new(RwLock::new(HashMap::new())),
+            tasks: Arc::new(RwLock::new(HashMap::new())),
+        })
+    }
+}
+
+/// Daphne-Worker per-request state.
+pub(crate) struct DaphneWorkerRequestState<'srv> {
+    pub(crate) shared_state: &'srv DaphneWorkerSharedState,
+
+    /// HTTP client to use for making requests.
+    pub(crate) client: reqwest_wasm::Client,
```

Comment:
Why not reuse this? It seems sensible to reuse any TLS/HTTP connections that we can.

Reply:
Yeah, it is safe to re-use after some research, so I restored it.